### PR TITLE
Allow spectral filtering of input data 

### PR DESCRIPTION
**Pull request by JuliaKukulies**
This pull request adds the spectral filtering feature (that has already been added to  `v2.0-dev` though #65) to `dev`.  The main changes I have made to the code are 

- to add the function `spectral_filtering()` in `utils.py`
- small changes in `feature_detection.py` to implement the filter step within the feature detection (same as for the optional gaussian filter)
- to add tests for the `spectral_filtering()` function itself and the filtering option in the feature detection function

For better comprehension what the function does, you can also check out the [example notebook](https://github.com/tobac-project/tobac-tutorials/blob/master/tutorials/Example_vorticity_tracking_model/Example_vorticity_tracking_model.ipynb)

### CODE REVIEW

**Comment by freemansw1** [comment]
Thanks @JuliaKukulies! I wonder if we should target a v1.4 release with these changes (and #138), given that the functionality is already in the v2.x branch and it seems to be well tested and less of a big change than #127 . I'm not sure. Either way, I'll try to get this reviewed next week.

**Comment by JuliaKukulies** [comment]
@freemansw1 Yes, I totally think we could that and have an intermediate release with this and #138. You are right that the changes are rather small and that it has already been reviewed for v.2 (just note that I have added a unit test that was not there before and that I made small changes in existing functions, so could be good to pay extra attention to that one more time).

What about #136 then? It is quite a lot of new code and a comprehensive new feature, so maybe rather something for 1.5 together with #127, but on the other hand it is also easier to review given that it is just an addition of code rather than changes in existing code. So could also be ready for 1.4 already, if we go for a 1.4 release.

File: tobac/tests/test_utils.py
Comment (freemansw1) [inline tobac/tests/test_utils.py:9]: Should we set a seed for this, or will any random values work?

**Review by freemansw1 — APPROVED** [review]
I just have a few comments, but I'm happy to approve/merge when we start staging for v1.4.

File: tobac/feature_detection.py
Comment (JuliaKukulies) [reply tobac/feature_detection.py:477]: Good point. I think we only discussed `dxy` and found it should be in m, so that it is the same unit as needed for the tracking. But I agree that I probably should have changed the wavelengths to m as well for consistency. To be honest I just did not think about this. It could be a bit more intuitive for users to give wavelengths in km because for most use cases they are probably ~ 100 000 - 10 000 000 m, but I am happy to change this if you think consistency is more important!

File: tobac/tests/test_utils.py
Comment (JuliaKukulies) [reply tobac/tests/test_utils.py]: I have resolved this by seeding a random wave signal with a wavelength outside of the set range for filtering. This can then be used to test the desired behavour of the spectral filtering: basically to remove this signal as it is not in the desired wavelength range.

File: tobac/feature_detection.py
Comment (JuliaKukulies) [reply tobac/feature_detection.py]: Good idea and agree - probably really best to have the same unit. Good that you took that up again, better to change this now than later! I require now to get the wavelengths in m as well and have changed the code where necessary. I have also added a warning as you suggested and a control that the value for minimum wavelength should be larger than `dxy`

**Comment by JuliaKukulies** [comment]
@freemansw1 I think I have addressed your two remaining points now. Sorry for making many commits, I had some issues with the tests, as I had to change the unit at more locations than I thought. Now everything should look fine. The main difference to the original PR are  

- changed required wavelength units to m instead of km  (this included also code changes where dxy was locally changed to get the same unit as the wavelengths, so actually the code should be cleaner now)
- modified test function for `spectral_filtering()` on data that actually has a wave signal rather than only any random data

**Review by freemansw1 — APPROVED** [review]
I'm happy with these changes now! We should wait to merge until we are staging for v1.4.0, but I am happy to merge otherwise.

**Comment by JuliaKukulies** [comment]
Great, thanks for your review! Yes, we wait for v1.4.0

**Comment by freemansw1** [comment]
I've moved this to merge into the new `RC_v1.4.0` now, so I'm happy for you to merge when ready @JuliaKukulies!

**Comment by JuliaKukulies** [comment]
All right, thanks @freemansw1! @w-k-jones do you think you wanna have a look as well or should I go ahead and merge?

**Comment by freemansw1** [comment]
Note with the merge of #138, auto-merging now fails with the documentation being at issue.

**Comment by JuliaKukulies** [comment]
I went ahead and merged this now after solving the merge conflict with #138, since you approved @freemansw1. I hope that was OK!

**Pull request merged**
